utils/crm_rpmcheck.py

```diff
@@ -36,36 +35,23 @@ def package_data(pkg):
 
     return {'name': pkg, 'error': "unknown package manager"}
 
+
 _packages = None
 def ansible_package_data(pkg) -> tuple[int, dict]:
     """
     Gathers version and release information about a package.
     Using ansible.
     """
+    # if _packages is None, then get it
     global _packages
     if not _packages:
-        # if _packages is None, then get it
-        rc, out, err = run(['ansible', '-m', 'package_facts', 'localhost'])
-        if rc == -1:
-            return -1, {}
-        # output format 'localhost | SUCCESS => { json...'
-        bracket_pos = out.find('{')
-        if bracket_pos == -1:
-            return -1, {}
-        is_ok = out[:bracket_pos].find('SUCCESS =>')
-        if is_ok == -1:
-            return -1, {}
-
-        # get the json part
-        out = out[bracket_pos:]
-        json_tree = json.loads(out)
-        # get _packages
-        _packages = json_tree['ansible_facts']['packages']
+        facts = utils.ansible_facts('package_facts')
+        _packages = facts['packages']
```

Comment:
Please use `.get` to access the item to avoid KeyError

Reply:
Refactored.